src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -88,6 +88,16 @@ public void save(AddressBook addressBook) throws StorageOperationException {
         /* Note: Note the 'try with resource' statement below.
          * More info: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html
          */
+    	try {
+    		File file = path.toFile();    	
+    		if (!file.exists() || file.isDirectory()){
+    			throw new FileNotFoundException();
+    		}
+    	}
+    	catch(FileNotFoundException fnfe){
+    		System.out.println("File not found, exiting");
+    		System.exit(0);
```

Comment:
as discussed in the tutorial, this may not be the most apt way to handle the exception; in this case the program halts. what you can do is go ahead and create the file anyway, and use the exception to inform user of the same.